index.coffee

```diff
@@ -118,9 +118,8 @@ class ProgressReporter
 	checkProgressError: (error, extraInfo) ->
 		prefix = "Progress error: [#{extraInfo}]"
 		console.warn(prefix, error.message ? error)
-		if error.message?.includes('unknown blob')
-			error.message = "#{prefix} #{error.message}"
-			throw error
```

Comment:
Instead of throwing only 'unknown blob' errors (and printing but otherwise ignoring any other errors), throw every error so that any progress error causes a build to fail. Logentries indicate that these errors are very rare -- I actually could not find any occurrence other than _"Cannot read property 'startsWith' of undefined",_ which is fixed in this PR.